src/core/address_space.cpp

```diff
@@ -52,18 +57,38 @@ struct AddressSpace::Impl {
         // to a reasonable amount.
         static constexpr size_t ReductionOnFail = 1_GB;
         static constexpr size_t MaxReductions = 10;
-        virtual_size = SystemSize + UserSize + ReductionOnFail;
+
+        system_managed_size = SystemManagedSize;
+        system_reserved_size = SystemReservedSize + ReductionOnFail;
+        user_size = UserSize;
         for (u32 i = 0; i < MaxReductions && !virtual_base; i++) {
-            virtual_size -= ReductionOnFail;
-            virtual_base = static_cast<u8*>(VirtualAlloc2(process, NULL, virtual_size,
-                                                          MEM_RESERVE | MEM_RESERVE_PLACEHOLDER,
-                                                          PAGE_NOACCESS, &param, 1));
+            system_reserved_size -= ReductionOnFail;
```

Comment:
Typically on windows the lower part of system_managed is reserved. And since this VirtualAlloc2 has address requirements which force it to the range we want, reducing the virtual size will most definitely cut system managed size and not reserved size. So you probably need to subtract this from system_managed_size